src/lib/auth.ts

```diff
@@ -130,15 +132,12 @@ export const getClassInfo = (classInfoUrl: string, rawPortalJWT: string) => {
   });
 };
 
-export const authenticate = (devMode: boolean) => {
+export const authenticate = (devMode: boolean, token?: string, domain?: string) => {
```

Comment:
I ended up moving the `urlParams` into parameters here instead of pulling them from the URL. This makes testing much easier, and is probably more like what we would want an eventual API to look like for an `auth` library, anyway